### ISSUE TITLE
Training Manager Default Repair Count

### ISSUE THREAD

**Issue by brycole** [rpherbig/dr-scripts#1785]
Tagging everyone from the original PRs.

@jandersson 
@rcuhljr 
@jayRyan24 
@rpherbig 

I noticed that a new repair counter was added here via #1755 and #1726 with a default of "1." This basically fubared anyone who was using the timer at a non-default value and caused them to immediately repair after every hunt.

For example, the default timer was 600 and my YAML was set to 14400.  When this change updated I immediately stopped using my user timed YAML and started repairing every hunt.  I think at the very least there should have been a notification that this value was being added in and to update your YAML.

I'm late to the PR discussion referenced above but I absolutely think a timer is more valuable because people's hunting times vary.  Some people do 25 minutes, some do 1 hour+.  A counter cannot distinguish between those two and is hard to predict.  

Basically, this default "1" behavior turns the repair settings back to what they were 6+ months ago, repair every loop.  It's really hard to predict a default "repair every" counter because hunting timers vary wildly. If we picked 1, then it could be 25 minutes or 2 hours. If we picked 5 it could be 10 hours or 2 hours, etc etc. 

I recommend we change the default counter to blank or 99 and let users set it as appropriate.  The default 10 minute duration timer will operate as a safe guard against never repairing.

Thanks!

**Comment by intangibles** [comment]
The wiki also needs to be updated with the new functionality.

**Comment by rcuhljr** [comment]
I haven't used any of these settings yet so I haven't been following this one super closely, but can't you just set your repair_every_count to 99 and you're back to the functionality you want? I can't find a strong reason to change the default behavior away from what it was originally, which was repairing after every hunt. The timer bit versus a counter is just personal preference, I know how long all of my hunts are and I repair appropriately.

**Comment by jandersson** [comment]
Believe blanking it out should work. I'm using the timer and have the counter set to infinity

**Comment by intangibles** [comment]
Repairing after every hunt is completely overkill.  Especially when you are starting out and testing out new yamls and such.  Wasting a good 5 minutes repairing things for 10 kronars is completely unnecessary.  Are people item's really getting destroyed in 1 hunt? I can't seem to find a single good reason for this.  If someone is hunting for 4-8 hours then they are the outlier and we shouldn't make defaults for them.

I can and have switched back to the timer method. My issue is that it forced everyone to use the new method w/o any notice and I also disagree with the new default behavior as noted above.

**Comment by izawwlgood** [comment]
There's a flag for it though? Why not just document the flag?

**Comment by rcuhljr** [comment]
We don't know how long people are going to hunt for. Not all items have the same base durability. Repairing each cycle has been the default up to now and is a reasonable safe option. The time loss is even more debatable when repair times go up with continued damage. You're talking maybe a minute  or two every 2-3 hours.

**Comment by intangibles** [comment]
Right, but if you leave the default at 10 minutes it removes all the irritation of having it repair every single time when you are testing or restarting your YAML.

Exactly, if you don't know how long people are going to hunt for, then repetition per X cycle is just as useless as X time.  But you can choose the one with the least annoyance and more control. e.g. Time.

I'm currently repairing every 4 hours, I could repair every 24 hours.  There's really never a time where repairing every hunt is mandatory, necessary, or beneficial.

**Comment by rcuhljr** [comment]
>Right, but if you leave the default at 10 minutes it removes all the irritation of having it repair every single time when you are testing or restarting your YAML.

Repairing happens after the run and can just be cancelled if you're testing or restarting, I can't say I've ever run into this situation as a real problem.

>I can't think of a single reason where repairing every hunt is mandatory, necessary, or beneficial.

Because it's the safer option for no real downside.

**Comment by rpherbig** [comment]
We've got a few people that do 1-4 hour hunts for various reasons.

**Comment by jandersson** [comment]
Repairing after each hunting-buddy iteration seems like the most sensible default to me. I have a feeling very many people don't change this.

**Comment by intangibles** [comment]
A few other things to think about too...

If you want to change your default to something other then "1 hunt cycle" you now need to change two settings.  Timer + counter.

We were also having a discussion in chat about "non-lichers" complaining in discord about all the scripting happening in crossing.  All the lichers tend to follow the same paths, same actions, etc, so its a bit easier to spot them and complain about them.  While "1 repair" may be optimal for some, we should be thinking about the impact on the game itself and whether we can take actions to appear less invasion.  

For example, if we up the repair count to 2 as default, is anything really lost?  On the other hand we stand to gain a lot on the whole by reducing the occurrence of repairs and thus the foot traffic, number of times at the repair shop, etc etc, which can reduce the appearance and invasive nature of scripting.

This is probably something we should start thinking about and incorporating in a lot of our scripts.

**Comment by jayRyan24** [comment]
It's not a detriment to have more options in this regard. 

```
repair_every: 3
repair_timer:
```

Is what I do. Scripting in general is a routine, theirs do it as much as ours. They just want to bitch about it because there's nothing else going on and we're the new thing. I could really give a rat's ass about the opinions of others in regards to the efficiency of our scripts.

**Comment by rpherbig** [comment]
I think we can consider things we might be doing that are causing problems. Waiting in line at Dokt is a good example of how we reacted to a problem.

That said, I haven't seen much evidence that repairing is causing a problem. We've been doing it this way for a long time with minimal complaint. And most of the complaints I've seen are "why aren't you using repair all?", which go away once we explain the limitations of `repair all`.

**Comment by intangibles** [comment]
@jayRyan24 
We absolutely have to acknowledge that Lich has made certain tasks easier.  The amount of effort it takes to spin up 10 F2P alts in genie compared to Lich is 100x. We absolutely SHOULD give a rat's ass about this. Granted, I'm not saying we should care about this happening, but we should absolutely try and take steps to minimize disruptive or repetitive behavior in the processes.

I wasn't around during the empath wars but I bet some of that could have been avoided by sacrificing a bit of efficiency for a little bit of grace.

@rpherbig 
Repairing isn't really the problem here. It's the amount of foot traffic and pathing that's generated.  Or lets say a new player is at the repair shop and notices like 50 people come in during that 2 minutes and repair without saying a single word.  Sure, not disruptive per se.  Annoying? Perhaps?  What if we could cut that down to 10 people?  Why wouldn't we?

The line at Dokt is a good example of efficiency colliding with common sense.  How many people get stuck there waiting for a scratch or tiny scuff? How many wait for 30+ minutes? I think most of the genie people I've talked to just lay down automatically because they know the lichers aren't going to respond. The behavior isn't rational or human and when you expand that behavior by 100 people it gets noticed and it looks like an army of bots.

I honestly think the main difference here is the similarities between a lot of the lich scripts. And when you again get 100+ people doing the same exact thing, the robotic nature becomes more apparent and just stands out.  And this isn't even an AFK thing.  Every single person could be at the keyboard and people would still complain.

And I know the difference because I used genie for 5+ years.  The goal of genie scripts were to be as unobtrusive as possible.  Lich flips that on its head and flaunts it. It's no wonder that people may find it irritating or abrasive.

**Comment by izawwlgood** [comment]
I don't think we should presume genie scripts are designed to be subtle.
The issue is that genie gets a pass with it because they all do it. To that
end, continued efforts to blend in are fine and good but we don't need to
go nuts.

I think setting the repair frequency default to 3-5 is a fine, but
ultimately the shit we're getting isn't going to be solved by continuing to
accommodate pissy genie users.

**Comment by rpherbig** [comment]
Isn't the sole purpose of Catrox and Randal to repair? Why would anyone think it odd if 10 people use the room for its intended purpose and then leave without saying anything? If that scales to 50 people, what is the problem since, unlike Dokt, multiple people can repair at the same time. Are people socializing in that room?

I'm not sure what you meant by 'empath wars' - if you are referring to the report and consent-baiting, then people went out of their way to report lichers. That wasn't because of any specific behaviors of our scripts, but because we were the favorite target that week of the trolls. The next week they moved on to harass others.

I'm not sure what to make of your Dokt comment. The line was _requested_ by non-lichers, so we implemented it. Combined with a 'minimum wound threshold' to skip healing scratches, Dokt is a lot less busy. That was after a conversation on the forums with several participants, one-on-one conversations (mostly over whispers), and so on.  That's a success story.

I feel like you are under the impression that there is a lot of annoyance at lichers. I don't get that impression. I feel like there are a vocal few who are annoyed, but always have been and always will be.

**Comment by rcuhljr** [comment]
```I think setting the repair frequency default to 3-5 is a fine, but
ultimately the shit we're getting isn't going to be solved by continuing to
accommodate pissy genie users.```

Ditto. I'm fine with 2 or 3 as a default. I don't think it was a real issue though.

**Comment by intangibles** [comment]
This sort of blossomed into a bigger discussion.  I agree, the repair issue by itself is probably not a real issue but It can certainly contribute.  It's when all the little things are added together that issues start to creep up.

Meh, I hear all kinds of stories and complaints. I've actually been lectured in game a few times for being one of "those people" who always asks if anyone is hunting in the room.

Anyways, even if this is is just one guy blowing smoke, being less intrusive isn't the end of the world.

**Comment by MahtraDR** [comment]
Is this still an issue? We can use more sensible defaults if called for. @rpherbig opinions? Old discussion, shall we try to resolve it, or close it?
